src/main/java/frc/trigon/robot/subsystems/arm/toohardarm/ToohardArmIO.java

```diff
@@ -0,0 +1,72 @@
+package frc.trigon.robot.subsystems.arm.toohardarm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import frc.trigon.robot.subsystems.arm.ArmIO;
+import frc.trigon.robot.subsystems.arm.ArmInputsAutoLogged;
+import frc.trigon.robot.utilities.Conversions;
+
+public class ToohardArmIO extends ArmIO {
+    private final CANSparkMax
+            masterAngleMotor = ToohardArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ToohardArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ToohardArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ToohardArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final TalonSRX elevatorEncoder = ToohardArmConstants.ELEVATOR_ENCODER;
+    private final double angleEncoderPositionSignal = ToohardArmConstants.ANGLE_ENCODER_POSITION_SIGNAL.getValue();
+    private final double angleEncoderVelocitySignal = ToohardArmConstants.ANGLE_ENCODER_VELOCITY_SIGNAL.getValue();
+
+    @Override
+    protected void updateInputs(ArmInputsAutoLogged inputs) {
+        inputs.angleMotorsVoltage = masterAngleMotor.getBusVoltage();
+        inputs.elevatorMotorsVoltage = masterElevatorMotor.getBusVoltage();
+
+        inputs.elevatorPositionRevolutions = Conversions.magTicksToRevolutions(elevatorEncoder.getSelectedSensorPosition());
+        inputs.elevatorVelocityRevolutionsPerSecond = Conversions.perHundredMsToPerSecond(Conversions.magTicksToRevolutions(elevatorEncoder.getSelectedSensorVelocity()));
+
+        inputs.angleEncoderPositionSignal = angleEncoderPositionSignal;
+        inputs.angleEncoderVelocitySignal = angleEncoderVelocitySignal;
+    }
+
+    @Override
+    protected void setTargetAngle(Rotation2d targetAngle) {
```

Comment:
This should accept a target velocity as well for the ff calculations. Use the `State` object.